frontend/src/components/Post.js

```diff
@@ -40,7 +40,9 @@ class Post extends React.Component {
     }
 
     onCommentPosted() {
-        this.commentListElement.current.loadComments();
+        //Whenver a comment is posted, inverse the boolean associated to refreshComment
+        //This state will be passed into a commentList in order to make the component rerender
+        this.setState({ refreshComment: !this.state.refreshComment })
```

Comment:
seems a little hacky to have a toggle. if we could change this in the future to something cleaner (i.e. only refresh when `refreshComment` becomes `true`) that would be better

Reply:
We could discuss about this, but I followed a tutorial and they suggested doing this, cause this would just indicate to the `commentList` component that a state has change, so it re-renders the component and fetch the updated component. Previously, you had used ref but I read that it wasn't recommended the way you used it.

Reply:
i agree that using `ref`'s was hacky as well but i feel that a toggle is worse. could you link me to the tutorial?

Reply:
when i was reading your code i had assumed that the `commentList` would update whenever `refreshComment` would be set to `true`. however, it will also refresh when `refreshComment` goes from `true` to `false` which seems counter-intuitive

Reply:
our current design is not good as currently there is no way to properly communicate from parent (`Post` component) to the child component (`CommentList` component). for instance, in the future, when we implement pagination, we might need to request more complex re-loading (ex: only reload the first 10 new comments). the current design is not really extensible (especially a toggle, because the child component does not differentiate a transition from `true` to `false` and a transition from `false` to `true`)

i think we should either:
- move the `PostComment` component inside the `CommentList` component (because child to parent communication would be easier in this case)
- let the `Post` handle the comment fetching, and instead of passing the post's `id` to the `CommentList` component, we pass an array of comments and make `CommentList` only handle the rendering of those comments

either way, we need to change this because eventually we will need to refactor it to make it work with future changes